### DISCUSSION TITLE
Add option to disable "Add new" button in `relationship` field

### DISCUSSION THREAD

**Discussion by 1nfinite9** [payloadcms/payload#1750] [Feature Requests & Ideas]
Currently, the admin panel presents a button to add an item to a collection on the fly on a `relationship` field.

<img width="730" alt="image" src="https://user-images.githubusercontent.com/4265044/209190885-0a348300-a1cc-4a10-bbf7-73ecea93b609.png">

There could be use cases where a user should be only limited to just creating the current item and not add additional item to a `relationship` collection, so it would be nice to have an option to disable this button.

**Comment by jacobsfletch** [comment]
@1nfinite9 that button does not show if the user lacks `create` access. It sounds like you're looking for something very similar. Can you add some access control to your users?

**Comment by 1nfinite9** [comment]
@jacobsfletch Apologies, I forgot to add that it would be nice to be able to do this without disabling it via access control. You'd still be able to create docs for that `relationship` individually, just not when editing or creating a doc that has a `relationship` to it.

**Comment by jacobsfletch** [comment]
@1nfinite9 totally, if we were to support this it would likely be a flag in somewhere in the config — something like `relationships.create: boolean | function`.

**Comment by 1nfinite9** [comment]
@jacobsfletch Thanks! That's aligned with what I was looking for. Would the `function` be for providing custom actions to the button?

**Reply by jacobsfletch** [reply]
It would behave very similarly to the existing field-level conditional logic: https://payloadcms.com/docs/fields/overview#conditional-logic

We would need some API to set conditions on only _parts_ of the field, as opposed to the whole thing.

**Comment by jacobsfletch** [comment]
@1nfinite9 I just opened a PR for this, check it out! https://github.com/payloadcms/payload/pull/2070
